src/utils/FileEmitter.ts

```diff
@@ -365,14 +365,23 @@ export class FileEmitter {
 		if (!fileName) {
 			const sourceHash = getSourceHash(source);
 			fileName = fileNamesBySource.get(sourceHash);
-			if (!fileName) {
-				fileName = generateAssetFileName(
-					consumedFile.name,
-					source,
-					sourceHash,
-					outputOptions,
-					bundle
-				);
+			const newFileName = generateAssetFileName(
+				consumedFile.name,
+				source,
+				sourceHash,
+				outputOptions,
+				bundle
+			);
+			// make sure file name deterministic in parallel emits, always use the shorter and smaller file name
+			if (
+				!fileName ||
+				fileName.length > newFileName.length ||
+				(fileName.length === newFileName.length && fileName.localeCompare(newFileName) > 0)
```

Comment:
Are you sure you want to use `localeCompare` here over just using `>`? My worry is that `localCompare` may
- make the output depend on the system language (in edge cases)
- can potentially determine two "different" strings to be the same, in which case again the output would not be deterministic (I think, though not 100% sure)

Reply:
Good point. I will switch to just use `>`